File: IPWorks Samples/Internet Paging/cpp/snpp.cpp
/*
 * IPWorks 2022 C++ Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworks
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */


#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "../../include/ipworks.h"

#define LINE_LEN 80

class MySNPP: public SNPP
{
public:
	bool waiting;

	virtual int FirePITrail(SNPPPITrailEventParams *e)
	{
		waiting = false;
		printf(  "%d  %s\n", e->Direction, e->Message );
		return 0;
	}
	virtual int FireError(SNPPErrorEventParams *e)
	{
		printf("Error %i: %s", e->ErrorCode, e->Description);
		exit(0);
	}
	virtual int FireSSLServerAuthentication(SNPPSSLServerAuthenticationEventParams *e)
	{
		if (e->Accept) return 0;
		printf("Server provided the following certificate:\nIssuer: %s\nSubject: %s\n",
			e->CertIssuer, e->CertSubject);
		printf("The following problems have been determined for this certificate: %s\n", e->Status);
		printf("Would you like to continue anyways? [y/n] ");
		if (getchar() == 'y') e->Accept = true;
		else exit(0);
		return 0;
	}
};


int main (int argc, char * argv[])
{
	if (argc != 4) {

		fprintf(stderr, "usage: snpp server pagerid message\n\n");
		fprintf(stderr, "  server   the name or address of the SNPP server\n");
		fprintf(stderr, "  pagerid  the identifying number of the pager to send a Message to\n");
		fprintf(stderr, "  message  the message content\n");
		fprintf(stderr, "\nExample: snpp SnppServer pagerId \"test message\"\n\n");
		printf("Press enter to continue.");
		getchar();

	}
	else{
		MySNPP snpp;
		
		snpp.SetServerName(argv[1]);

		snpp.SetPagerId(argv[2]);

		snpp.SetMessage(argv[3]);

		snpp.Connect();
		if (!snpp.GetConnected())
		{
			printf("Cannot connect to server. Press enter to continue...\n");
			getchar();
			exit(0);
		}
		snpp.Send();

		printf("Page sent. Press enter to continue...");
		getchar();
		exit(1);
		return 0;
	}	
}

File: IPWorks Samples/Internet Paging/java/snpp.java
/*
 * IPWorks 2022 Java Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworks
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */

import java.io.*;
import ipworks.*;

public class snpp extends ConsoleDemo {
	
	
	public static void main(String[] args) {
		
		if(args.length!=3) {
			
			System.out.println("usage: snpp server pagerid message");
			System.out.println("");
			System.out.println("  server   the name or address of the SNPP server");			
			System.out.println("  pagerid  the identifying number of the pager to send a Message to");			
			System.out.println("  message  the message content");
			System.out.println("\r\nExample: snpp SnppServer pagerId \"test message\"");
		
		} else {
			
			Snpp snpp1 = new Snpp();
			
			try {
				snpp1.addSnppEventListener(new DefaultSnppEventListener(){
					
					public void PITrail(SnppPITrailEvent e) {		
						System.out.println(e.direction + "  " + e.message);	
					}					
					public void SSLServerAuthentication(SnppSSLServerAuthenticationEvent e) {
						e.accept=true; //this will trust all certificates and it is not recommended for production use	
					}					
					public void error(SnppErrorEvent e) {
						System.out.println("Error " + e.errorCode + ": " + e.description);		
					}
									
		        });
					
				snpp1.setServerName(args[0]);
				snpp1.setPagerId(args[1]);
				snpp1.setMessage(args[2]);

				snpp1.connect();
				snpp1.send();
				System.out.println("Page sent.");
			} catch (Exception e) {
				displayError(e);
			}

			try {
				snpp1.disconnect();
			} catch (Exception e) {
				displayError(e);
				System.exit(1);
			}
		}			
	}
}

class ConsoleDemo {
  private static BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

  static String input() {
    try {
      return bf.readLine();
    } catch (IOException ioe) {
      return "";
    }
  }
  static char read() {
    return input().charAt(0);
  }

  static String prompt(String label) {
    return prompt(label, ":");
  }
  static String prompt(String label, String punctuation) {
    System.out.print(label + punctuation + " ");
    return input();
  }

  static String prompt(String label, String punctuation, String defaultVal)
  {
	System.out.print(label + " [" + defaultVal + "] " + punctuation + " ");
	String response = input();
	if(response.equals(""))
		return defaultVal;
	else
		return response;
  }

  static char ask(String label) {
    return ask(label, "?");
  }
  static char ask(String label, String punctuation) {
    return ask(label, punctuation, "(y/n)");
  }
  static char ask(String label, String punctuation, String answers) {
    System.out.print(label + punctuation + " " + answers + " ");
    return Character.toLowerCase(read());
  }

  static void displayError(Exception e) {
    System.out.print("Error");
    if (e instanceof IPWorksException) {
      System.out.print(" (" + ((IPWorksException) e).getCode() + ")");
    }
    System.out.println(": " + e.getMessage());
    e.printStackTrace();
  }
}



